lib/generators/apple_sign_in/config/devise_generator.rb

```diff
@@ -0,0 +1,12 @@
+module AppleSignIn
+  module Generators
+    class DeviseGenerator < Rails::Generators::Base
+      source_root File.expand_path('templates', __dir__)
+
+      def copy_devise_file
+
+        copy_file 'apple_sing_in_controller.rb', "#{source}apple_sing_in_controller.rb"
```

Comment:
I think you are missing to define `source`
You can do it by adding next to your class
```
argument :source
```